src/ThreeEditor/js/object/Object.Dimensions.ts

```diff
@@ -0,0 +1,354 @@
+
+		this.panel.add(
+			this.typeRow,
+			this.typeSelectRow,
+			this.xLengthRow,
+			this.yLengthRow,
+			this.zLengthRow,
+			this.radiusRow,
+			this.radiusRow2,
+			this.zoneIdRow
+		);
+	}
+
+	private setGeometryType(object: BasicMesh | DetectGeometry | WorldZone): string | undefined {
+		let geometryType;
+		if (isBasicMesh(object)) {
+			showUIElement(this.typeRow);
+			hideUIElement(this.typeSelectRow);
+			this.type.setValue(object.geometryType);
+			geometryType = object.geometryType;
+		} else {
+			hideUIElement(this.typeRow);
+			showUIElement(this.typeSelectRow);
+			if (isWorldZone(object)) {
+				this.typeSelect.setOptions(BASIC_GEOMETRY_OPTIONS);
+				this.typeSelect.setValue(object.geometryType);
+				geometryType = object.geometryType;
+			} else if (isDetectGeometry(object)) {
+				this.typeSelect.setOptions(DETECT_OPTIONS);
+				this.typeSelect.setValue(object.detectType);
+				geometryType = object.detectType;
+			}
+		}
+		return geometryType;
+	}
+
+	private setBox(object: BasicMesh | DetectGeometry | WorldZone): void {
+		showUIElement(this.xLengthRow);
+		showUIElement(this.yLengthRow);
+		showUIElement(this.zLengthRow);
+		if (isWorldZone(object)) {
+			if (object.autoCalculate) {
+				hideUIElement(this.xLengthRow);
+				hideUIElement(this.yLengthRow);
+				hideUIElement(this.zLengthRow);
+			} else {
+				const { size } = object;
+				this.xLength.setValue(size.x);
+				this.yLength.setValue(size.y);
+				this.zLength.setValue(size.z);
+			}
+		} else {
+			let parameters;
+			if (isDetectGeometry(object)) parameters = object.geometryData;
+			else if (isBoxMesh(object)) parameters = object.geometry.parameters;
+			else return;
+			this.xLength.setValue(parameters.width);
+			this.yLength.setValue(parameters.height);
+			this.zLength.setValue(parameters.depth);
+		}
+	}
+
+	private setSphere(object: BasicMesh | DetectGeometry | WorldZone): void {
+		showUIElement(this.radiusRow);
+		if (isWorldZone(object)) {
+			const { size } = object;
+			this.radius.setValue(size.x);
+		} else {
+			let parameters;
+			if (isDetectGeometry(object)) parameters = { radius: object.geometryData.radius };
+			else if (isSphereMesh(object)) parameters = object.geometry.parameters;
+			else return;
+			this.radius.setValue(parameters.radius);
+		}
+	}
+
+	private setCylinder(object: BasicMesh | DetectGeometry | WorldZone): void {
+		showUIElement(this.zLengthRow);
+		showUIElement(this.radiusRow);
+		if (isWorldZone(object)) {
+			const { size } = object;
+			this.radius.setValue(size.x);
+			this.zLength.setValue(size.z);
+		} else if (isDetectGeometry(object)) {
+			showUIElement(this.radiusRow2);
+			const parameters = object.geometryData;
+			this.radius.setValue(parameters.radius);
+			this.radius2.setValue(parameters.innerRadius);
+			this.radius.min = parameters.innerRadius + 0.001; // Prevent radius from being lower than innerRadius
+			this.radius2.max = parameters.radius - 0.001; // innerRadius cannot be greater than radius
```

Comment:
```suggestion
			this.radius.min = parameters.innerRadius + 1e-12; // Prevent radius from being lower than innerRadius
			this.radius2.max = parameters.radius - 1-12; // innerRadius cannot be greater than radius
```

What about much smaller values of threshold ?

Reply:
UI won't display it. There is a special issue that is supposed to enable it.
#199

Reply:
OK, thanks !

Reply:
I can add it now but doesn't look right.
![image](https://user-images.githubusercontent.com/26445545/144509448-8444a00c-c311-49ac-8f80-f115a503db29.png)

Reply:
If we go below 2e-5 the visualization starts to break
![image](https://user-images.githubusercontent.com/26445545/144511050-eb759c0f-4b55-494b-b5d7-fa2fff2cdf03.png)
but I doubt anyone  would need to make such a weird geometry for detects

Reply:
We can stay with larger values. The same can be fixed with global unit change, i.e. cm -> um or nm.
But this would go beyond the scope of current PR.
I.d recommend to display number with 4 digits after dot now.